dbt/adapters/spark/connections.py

```diff
@@ -409,9 +412,8 @@ def open(cls, connection):
                     cls.validate_creds(creds, required_fields)
 
                     dbt_spark_version = __version__.version
-                    user_agent_entry = (
-                        f"dbt-labs-dbt-spark/{dbt_spark_version} (Databricks)"  # noqa
-                    )
+                    dbt_invocation_env = os.getenv(DBT_INVOCATION_ENV) or "manual"
+                    user_agent_entry = f"dbt-labs-dbt-spark/{dbt_spark_version} (Databricks) ({dbt_invocation_env})"  # noqa
```

Comment:
Integration tests don't like this!
```
Runtime Error
E                   ('HY000', "[HY000] [unixODBC][Simba][DriverSupport] (2021) Invalid character ')' found in user-agent comment. (2021) (SQLDriverConnect)")
```
@ueshin Any chance you could advise here? This is an ODBC user agent, maybe the two sets of parens aren't allowed?

Reply:
Yeah, seems like they are not allowed for now. I'm not sure it's a bug or expected, though.
How about using a different pattern like `f"dbt-labs-dbt-spark/{dbt_spark_version} (Databricks; {dbt_invocation_env})"`?